### PR TITLE
Add javascript documentation for Crons upsert

### PR DESCRIPTION
**Pull request by gaprl**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Adds Crons upsert documentation to all javascript projects.

### CODE REVIEW

File: src/includes/javascript-crons-upsert.mdx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  },
+  {
+    schedule: {
+      type: "crontab",
+      value: "* * * * *",
+    },
+    checkinMargin: 2, // In minutes. Optional.
+    maxRuntime: 10, // In minutes. Optional.
+    timezone: "America/Los_Angeles", // Optional.
+  }
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:19]: Would it make sense to define monitorConfig separately and then pass it in, as done in the next code sample? I think this would make it clearer what this sample is showing.

File: src/includes/javascript-crons-upsert.mdx [hunk 2]
```diff
@@ -0,0 +1,124 @@
+You can create and update your Monitors programmatically with code rather than [creating and configuring them in Sentry.io](https://sentry.io/crons/create/).
+
+Using `Sentry.withMonitor()`, you can pass a third parameter with your monitor configuration:
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:3]: Is this creating the monitor? Can we reword this to be more action-oriented?

```suggestion
To create/update a monitor, use `Sentry.withMonitor()` and pass in your monitor configuration as a third parameter:
```

File: src/includes/javascript-crons-upsert.mdx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      value: "* * * * *",
+    },
+    checkinMargin: 2, // In minutes. Optional.
+    maxRuntime: 10, // In minutes. Optional.
+    timezone: "America/Los_Angeles", // Optional.
+  }
+);
+```
+
+Using `Sentry.captureCheckIn()`, you can pass a second parameter with your monitor configuration:
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:23]:
```suggestion
To configure the monitor's check-ins, use `Sentry.captureCheckIn()` and pass in your monitor configuration as a second parameter:
```

File: src/includes/javascript-crons-upsert.mdx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    // Make sure this variable is named `checkInId`
+    checkInId,
+    monitorSlug: "<monitor-slug>",
+    status: "ok",
+  },
+  monitorConfig
+);
+```
+
+Available monitor config properties:
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:95]:
```suggestion
### Monitor Configuration Properties

 The following are available monitor configuration properties:
```

File: src/includes/javascript-crons-upsert.mdx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+`schedule`:
+
+: The job's schedule :
+
+The schedule representation for your monitor, either `crontab` or `interval`. The structure will vary depending on the type:
+
+```json
+{"type": "crontab", "value": "0 * * * *"}
+{"type": "interval", "value": "2", "unit": "hour"}
+```
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:106]:
```suggestion
  The schedule representation for your monitor, either `crontab` or `interval`. The structure will vary depending on the type:
  
  ```json
  {"type": "crontab", "value": "0 * * * *"}
  {"type": "interval", "value": "2", "unit": "hour"}
  ```
```

Small nit, but can we tab these in a bit so the info organization is visually clearer?

**Review by vivianyentran — COMMENTED** [review]
Overall looks good! Added some suggestions.

**Review by AbhiPrasad — APPROVED** [review]
code changes looks good!

File: src/includes/javascript-crons-upsert.mdx (hunk 1 above)
Comment (gaprl) [reply src/includes/javascript-crons-upsert.mdx]: Done

File: src/includes/javascript-crons-upsert.mdx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+`schedule`:
+
+: The job's schedule :
+
+The schedule representation for your monitor, either `crontab` or `interval`. The structure will vary depending on the type:
+
+```json
+{"type": "crontab", "value": "0 * * * *"}
+{"type": "interval", "value": "2", "unit": "hour"}
+
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:132]:
```suggestion
```

```
Looks like this code block ending got moved

File: src/includes/javascript-crons-upsert.mdx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+</Note>
+
+`maxRuntime`:
+
+: The amount of time (in minutes) your job is allowed to run before it's considered failed. Optional.
+
+`timezone`
+
+: The `tz` where your job is running. This is usually your server's timezone, (such as `America/Los_Angeles`). See [list of tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones). Optional.
+```
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:150]:
```suggestion
```

File: src/includes/javascript-crons-upsert.mdx [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+We recommend that your check-in margin be less than or equal to your interval.
+
+</Note>
+
+`maxRuntime`:
+
+: The amount of time (in minutes) your job is allowed to run before it's considered failed. Optional.
+
+`timezone`
```
Comment (vivianyentran) [inline src/includes/javascript-crons-upsert.mdx:147]:
```suggestion
`timezone`:
```

**Review by vivianyentran — COMMENTED** [review]
some edits to fix formatting

**Review by vivianyentran — APPROVED** [review]
looks good, thanks for updating the docs!

**Comment by gaprl** [comment]
@AbhiPrasad done, we can update for Bun if you'd like as well!
